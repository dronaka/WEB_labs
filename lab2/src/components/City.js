import React from "react";


import WeatherParam from "./WeatherParam";

import Loader from "./Loader";
import "./styles/City.css";




export default class City extends React.Component {
  componentDidMount() {
    let forecast = this.props.onFetch();
    console.log("tyt")
    console.log(forecast)
  }

  render() {
    if (!this.props.forecast) {
      return this.renderLoader();
    }

    return this.renderWeather();
  }

  renderLoader() {
    return <Loader />
  }

  renderWeather() {
    const {
      forecast: {
        cityName,
        temperature,
        windSpeed,
        description,
        pressure,
        humidity,
        coords: {
          lat: latitude,
          lon: longitude
        } = {}
      },
      onDelete
    } = this.props;

    return (
      <div className="city">
        <div className="header">
          <div className="city-name">{cityName}</div>
          <div className="temperature">{temperature} &#8451;</div>
        </div>

        <WeatherParam name="Wind" value={`${windSpeed} m/s`} />
        <WeatherParam name="Cloudness" value={description} />
        <WeatherParam name="Pressure" value={`${pressure} hPa`} />
        <WeatherParam name="Humidity" value={`${humidity}%`} />
        <WeatherParam name="Coords" value={`${latitude}, ${longitude}`} />

        {onDelete && <button className="button" onClick={onDelete}>delete</button>}
      </div>
    );
  }
}