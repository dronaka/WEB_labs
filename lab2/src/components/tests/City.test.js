import React from "react";
import renderer from "react-test-renderer";

import City from "../City";

describe("<City /> render", () => {

  const DEFAULT_FORECAST = {
    cityName: "Moscow",
    temperature: "-4",
    icon: "icon_code",
    windSpeed: 10,
    description: "nice to swim",
    pressure: 234,
    humidity: 99,
    coords: {
      lat: 40,
      lon: 40
    }
  };

  test("should has forecast and button delete", () => {
    const tree = renderer
      .create(
        <City
          onFetch={() => {}}
          onDelete={() => {}}
          forecast={DEFAULT_FORECAST}
        />)
      .toJSON();
    
    expect(tree).toMatchSnapshot();
  });
  

  test("should has <Loader />", () => {
    const tree = renderer
      .create(
        <City
          onFetch={() => {}}
        />)
      .toJSON();
    
    expect(tree).toMatchSnapshot();
  });

});