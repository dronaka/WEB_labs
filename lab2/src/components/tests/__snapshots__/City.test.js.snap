// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<City /> render should has <Loader /> 1`] = `
<div
  className="loader"
>
  <p>
    Forecast is loading...
  </p>
</div>
`;

exports[`<City /> render should has <Loader /> even if onDelete given 1`] = `
<div
  className="loader"
>
  <p>
    Forecast is loading...
  </p>
</div>
`;

exports[`<City /> render should has forecast and no delete button 1`] = `
<div
  className="city"
>
  <div
    className="header"
  >
    <div
      className="city-name"
    >
      Moscow
    </div>
    <div
      className="temperature"
    >
      -4
       ℃
    </div>
    <img
      alt="Weather icon"
      src="https://openweathermap.org/img/wn/icon_code.png"
    />
  </div>
  <div
    className="param"
  >
    <div
      className="name"
    >
      Wind
    </div>
    <div
      className="value"
    >
      10 m/s
    </div>
  </div>
  <div
    className="param"
  >
    <div
      className="name"
    >
      Cloudness
    </div>
    <div
      className="value"
    >
      nice to swim
    </div>
  </div>
  <div
    className="param"
  >
    <div
      className="name"
    >
      Pressure
    </div>
    <div
      className="value"
    >
      234 hPa
    </div>
  </div>
  <div
    className="param"
  >
    <div
      className="name"
    >
      Humidity
    </div>
    <div
      className="value"
    >
      99%
    </div>
  </div>
  <div
    className="param"
  >
    <div
      className="name"
    >
      Coords
    </div>
    <div
      className="value"
    >
      40, 40
    </div>
  </div>
</div>
`;

exports[`<City /> render должна быть инфа о погоде и кнопка delete 1`] = `
<div
  className="city"
>
  <div
    className="header"
  >
    <div
      className="city-name"
    >
      Moscow
    </div>
    <div
      className="temperature"
    >
      -4
       ℃
    </div>
    <img
      alt="Weather icon"
      src="https://openweathermap.org/img/wn/icon_code.png"
    />
  </div>
  <div
    className="param"
  >
    <div
      className="name"
    >
      Wind
    </div>
    <div
      className="value"
    >
      10 m/s
    </div>
  </div>
  <div
    className="param"
  >
    <div
      className="name"
    >
      Cloudness
    </div>
    <div
      className="value"
    >
      nice to swim
    </div>
  </div>
  <div
    className="param"
  >
    <div
      className="name"
    >
      Pressure
    </div>
    <div
      className="value"
    >
      234 hPa
    </div>
  </div>
  <div
    className="param"
  >
    <div
      className="name"
    >
      Humidity
    </div>
    <div
      className="value"
    >
      99%
    </div>
  </div>
  <div
    className="param"
  >
    <div
      className="name"
    >
      Coords
    </div>
    <div
      className="value"
    >
      40, 40
    </div>
  </div>
  <button
    className="button"
    onClick={[Function]}
  >
    delete
  </button>
</div>
`;
